Add note on generic client alternative (#274)
SAP/cloud-sdk: docs/java/features/odata/type-safe-client-odata-v2.mdx

The Java classes represent the _data model_ and the available _operations_ of the service.
As a consequence all requests that are built through the typed OData v2 client are not only _syntactically valid_ but also _semantically valid_.

> :::info

Reviewer: ```suggestion
:::info Generic OData Client
```

> The type safe client may in some situations not be suitable for your needs.

Reviewer: ```suggestion
The type-safe client by the SAP Cloud SDK may in some situations not be suitable for your needs.
```

For instance a service might not fully conform to the OData standard, forcing you to perform customized requests.

> Or the typed API doesn't support a specific feature just yet.

Reviewer: ```suggestion
Or the typed API doesn't support a specific OData feature just yet.
```

> For such cases you can leverage the more [low-level generic OData Client](./generic-untyped-odata-client).

Reviewer: ```suggestion
For such cases, we recommend leveraging our [low-level generic OData Client](./generic-untyped-odata-client) or fallback directly to HTTP Client level if that's not enough.
In all the other situations typed client is preferable and considered the best practice. 
```
